### ISSUE TITLE
Use bool instead of int where applicable

### ISSUE THREAD

**Issue by pgaskin** [cmus/cmus#1020]
While working on my PRs and testing the others, I noticed there's a mix of `bool`s, `int`s, and `: 1` bitfields.

@flyingmutant, would you want me to work on refactoring these?

**Comment by flyingmutant** [comment]
I'd say in general cmus' codebase is quite an, lets say, interesting thing :-) Cleanup PRs are OK i guess, but be aware that important things can require more major refactorings -- and I don't think I'll be able to review those properly.

**Comment by pgaskin** [comment]
> Cleanup PRs are OK i guess, but be aware that important things can require more major refactorings -- and I don't think I'll be able to review those properly.

Yep, I'm aware of that. That's why I'm not planning on doing things like changing the mess of `#define`s into enums - there's a good chance of breaking something inadvertently since funny stuff is done with them like adding the value of one to another to modify it.

I'll probably be primarily focusing on the safest lowest-cost changes with the greatest effect on readability and reducing the possibility of bugs. For example, for this, the `int`s made it take quite a bit longer to get familiar with the code, made it harder to reason about the impact of other changes, was quite error-prone, and required going back and forth in the code quite a bit.

**Comment by pgaskin** [comment]
Note that I probably won't be working on this until after the next release since there is a chance of breakage, and I'd rather focus on the other issues for now.
